### ISSUE TITLE
Comment On/Off Feature

### ISSUE THREAD

**Issue by AmeyaBhujbal** [oracle/opengrok#3584]
I am using OpenGrok 1.7.2
I would like to request you to add a new feature of **Comment On/Off**
Usually, Opengrok searches for a keyword and shows the results having that keyword **(even if it is commented out in source code)**
Sometimes, because of which it gets really difficult to find the correct match one is looking for.

It'd be really helpful if we have **Comment On/Off** option. Wherein, if we opt for **Comment Off** option, opengrok wouldn't show the result that is commented out in the source code. So, it would be easier to look for the correct match.

Also wanted to confirm if it is added in the new OpenGrok 1.7.3?

**Comment by vladak** [comment]
How do you perform the search exactly ? Is it full text / symbol / definition search ? Can you post an example of the code ?

**Comment by AmeyaBhujbal** [comment]
It is full search.

Example below:
Searched for **geometry** in FULL SEARCH -

34 //  removal of UPPERCASE geometry
86 PPG_FACE(class surface *geometry/*=NULL*/, REVBIT sense/*=FORWARD*/,
89 PPG_FACE(class surface *geometry/*=NULL*/);
282 //return the surface geometry (a surface)
289 //set the surface geometry
306 inline PPG_FACE::PPG_FACE(class surface *geometry, REVBIT sense, in PPG_FACE() argument
308 : FACE(geometry, sense, type) in PPG_FACE()
327 inline PPG_FACE::PPG_FACE(class surface *geometry, REVBIT sense) in PPG_FACE() argument
328 : FACE(geometry, sense, PPG_FACE::type) in PPG_FACE()
336 inline PPG_FACE::PPG_FACE(class surface *geometry) in PPG_FACE() argument

if you see line no. 34, 282 & 289, these lines are commented out

**Comment by vladak** [comment]
Yes, the full search performs search in all the tokens, including those the comments. If you want just references or definitions, use Definition/Symbol search.

**Comment by AmeyaBhujbal** [comment]
References or definitions search has a limited scope. I want full search without comments

**Comment by vladak** [comment]
What do you mean by "limited scope" ?

**Comment by AmeyaBhujbal** [comment]
It does not search in string if we have set it to references or definitions

**Comment by vladak** [comment]
What do you mean by 'search in string' exactly ? Maybe give an example ?
